projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -1135,6 +1137,16 @@ function ( $category_id ) {
 
 					break;
 
+				case 'in_site_migration_flow':
+					if ( ! (bool) $value ) {
+						delete_option( 'in_site_migration_flow' );
+					} else {
+						update_option( 'in_site_migration_flow', (int) (bool) $value );
+					}
+
+					$updated[ $key ] = (int) (bool) $value;
```

Comment:
Minor thought: rather than repeating the casts multiple times, might it make sense to cast and then do all the logic?
```suggestion
					$canonical_value = (int) (bool) $value;
					if ( 0 === $canonical_value ) {
						delete_option( 'in_site_migration_flow' );
					} else {
						update_option( 'in_site_migration_flow', $canonical_value );
					}

					$updated[ $key ] = $canonical_value;
```